schema/C2_LOAN_GUARANTOR.json

```diff
@@ -1,412 +1,459 @@
-            }
-        },
-        "guarantor_suffix": {
-            "title": "GUARANTOR SUFFIX",
-            "description": "",
-            "type": [
-                "string",
-                "null"
-            ],
-            "minLength": 0,
-            "maxLength": 10,
-            "pattern": "^[ -~]{0,10}$",
-            "examples": [
-                "Jr"
-            ],
-            "fec_spec": {
-                "COL_SEQ": 9,
-                "FIELD_DESCRIPTION": "GUARANTOR SUFFIX",
-                "TYPE": "A/N-10",
-                "SAMPLE_DATA": "Jr"
-            }
+            "maxLength": 20
+          }
+        }
+      }
+    },
+    {
+      "if": {
+        "properties": {
+          "entity_type": {
+            "const": "COM"
+          }
         },
-        "guarantor_street_1": {
-            "title": "GUARANTOR STREET 1",
-            "description": "",
+        "required": ["entity_type"]
+      },
+      "then": {
+        "required": [
+          "guarantor_organization_name",
+          "guarantor_committee_fec_id"
+        ],
+        "properties": {
+          "guarantor_organization_name": {
             "type": "string",
             "minLength": 1,
-            "maxLength": 34,
-            "pattern": "^[ -~]{0,34}$",
-            "examples": [
-                "123 Main Street"
-            ],
-            "fec_spec": {
-                "COL_SEQ": 10,
-                "FIELD_DESCRIPTION": "GUARANTOR STREET 1",
-                "TYPE": "A/N-34",
-                "REQUIRED": "X (error)",
-                "SAMPLE_DATA": "123 Main Street"
-            }
-        },
-        "guarantor_street_2": {
-            "title": "GUARANTOR STREET 2",
-            "description": "",
-            "type": [
-                "string",
-                "null"
-            ],
-            "minLength": 0,
-            "maxLength": 34,
-            "pattern": "^[ -~]{0,34}$",
-            "fec_spec": {
-                "COL_SEQ": 11,
-                "FIELD_DESCRIPTION": "GUARANTOR STREET 2",
-                "TYPE": "A/N-34"
-            }
-        },
-        "guarantor_city": {
-            "title": "GUARANTOR CITY",
-            "description": "",
+            "maxLength": 200
+          },
+          "guarantor_committee_fec_id": {
             "type": "string",
-            "minLength": 1,
-            "maxLength": 30,
-            "pattern": "^[ -~]{0,30}$",
-            "examples": [
-                "Anytown"
-            ],
-            "fec_spec": {
-                "COL_SEQ": 12,
-                "FIELD_DESCRIPTION": "GUARANTOR CITY",
-                "TYPE": "A/N-30",
-                "REQUIRED": "X (error)",
-                "SAMPLE_DATA": "Anytown"
-            }
+            "minLength": 9,
+            "maxLength": 9
+          }
+        }
+      }
+    },
+    {
+      "if": {
+        "properties": {
+          "guaranteed_amount": {
+            "type": "number",
+            "minimum": 200.01
```

Comment:
I think this should also only apply if the entity type is IND, since otherwise it might prevent COM transactions from saving